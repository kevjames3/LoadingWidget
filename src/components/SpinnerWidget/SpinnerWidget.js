import React, { Component } from "react";
import "./styles/widget.css";

class SpinnerWidget extends Component {
  constructor(props) {
    super(props);
    this.maxFill = 158;
  }
  render() {
    return (
      <div className="spinner_widget">
        <svg
          height="100%"
          width="100%"
          viewBox="0 0 100 100"
          preserveAspectRatio="none"
          className={this.props.spinning === true ? "animate" : "pause-animate"}
        >
          <circle
            className="outer_circle"
            r="25"
            cx="50"
            cy="50"
            style={{
              strokeDasharray: `${Math.ceil(
                (this.props.value / 100) * 158
              )} 158`,
            }}
          />
          <circle className="inner_circle" r="45" cx="50" cy="50" />
        </svg>
        <div className="status">{this.props.value}%</div>
      </div>
    );
  }
}

export default SpinnerWidget;
