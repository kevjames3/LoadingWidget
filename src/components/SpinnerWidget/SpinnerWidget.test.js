import React from "react";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import SpinnerWidget from "./SpinnerWidget";

afterEach(cleanup);

describe("SpinnerWidget", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SpinnerWidget />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the default value correctly", () => {
    const { getByText } = render(<SpinnerWidget />);
    expect(getByText(/%/i).textContent).toBe("0%");
  });

  it("renders the correct value and the stroke array is correct", () => {
    const { getByText, container } = render(<SpinnerWidget value={55} />);
    expect(getByText(/%/i).textContent).toBe("55%");
    expect(container.querySelector(`circle.outer_circle`)).toHaveStyle(
      "stroke-dasharray: 87 158"
    );
  });

  it("animates when it is told not to", () => {
    const { container } = render(<SpinnerWidget value={55} spinning={false} />);
    expect(container.querySelector("svg.pause-animate")).toBeTruthy();
  });

  it("animates when it is told to", () => {
    const { container } = render(<SpinnerWidget value={55} spinning={true} />);
    expect(container.querySelector("svg.animate")).toBeTruthy();
  });
});
