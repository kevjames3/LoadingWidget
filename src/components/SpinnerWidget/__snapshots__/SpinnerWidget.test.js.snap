// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinnerWidget renders correctly 1`] = `
<div
  className="spinner-widget"
>
  <svg
    className="pause-animate"
    height="100%"
    preserveAspectRatio="none"
    viewBox="0 0 100 100"
    width="100%"
  >
    <circle
      className="outer_circle"
      cx="50"
      cy="50"
      r="25"
      style={
        Object {
          "strokeDasharray": "0 158",
        }
      }
    />
    <circle
      className="inner_circle"
      cx="50"
      cy="50"
      r="45"
    />
  </svg>
  <div
    className="status"
  >
    0
    %
  </div>
</div>
`;
